pallets/creditcoin/src/migrations/mod.rs

```diff
@@ -44,3 +44,34 @@ pub(crate) fn migrate<T: Config>() -> Weight {
 
 	weight
 }
+
+macro_rules! projection_alias {
+	($($config: path : { $($typ: ident),+ }),+) => {
+		$(
+			$(
+				paste::paste! {
+					type [<$typ Of>]<T> = <T as $config>::$typ;
+				}
+			)+
+		)+
+	};
+}
+
+projection_alias!(frame_system::Config: { Hash, BlockNumber, AccountId }, pallet_timestamp::Config: { Moment });
+
+macro_rules! storage_macro {
+	($name: ident, $t: ident, $storage: ident < $($typ: ty),+ >) => {
+		paste::paste! {
+			macro_rules! [<$name:snake _storage>] {
+				($inner_t: ident, $thing: ty) => {
+					#[frame_support::storage_alias]
+					type $name<$inner_t: crate::Config> = $storage < $($typ),+ , $thing >;
+				};
+			}
+			#[allow(unused_imports)]
+			use [<$name:snake _storage>];
+		}
+	};
+}
+
+use storage_macro;
```

Comment:
Yeah, the difference is that `#[macro_export]`
1. Always exports the macro from the crate root
2. Doesn't obey normal visibility rules (it's always externally public from the crate root)

This is a trick to get declarative macros to obey normal visibility/scoping rules.
(Pretty much declarative macros exist in a separate namespace, treated differently from normal items, due to backwards compatibility. The macro definition is always usable after the definition, within the same file, though, so you can refer to it by name. And since `use` declarations obey normal visibility rules, if you reexport the macro with a `use` declaration you can get the normal behavior. [More info](https://stackoverflow.com/a/67140319/19453822))